README.plugin.md

```diff
@@ -7,6 +7,18 @@
 See [How_to_Install_a_Plugin](http://projects.theforeman.org/projects/foreman/wiki/How_to_Install_a_Plugin)
 for how to install Foreman plugins
 
+## Migrating your plugin:
```

Comment:
> The wiki's the main source of detailed maintainer docs.

Thanks, I'll put it there.

> this probably doesn't implement the complete process

Actually it's all that is needed - first remove the migrations as I mentioned and then you can remove the code dependency just as you have added it be it RPM, code or gem.
There are still pitfalls like "orphaned" settings e.t.c. but they should be handled by the plugins IMHO.
My only consideration is the way the user consumes the plugin: if the plugin is packaged - it should be possible to add a step that will perform the migration automatically, in any other case  (gem or code) the user is responsible for running the migration.